src/replication.c

```diff
@@ -3327,13 +3451,14 @@ cleanup_deadlist(void *arg)
 
 			ASSERT(rcomm_cmd->state == CMD_EXECUTION_DONE);
 
-			for (i = 0; i < rcomm_cmd->copies_sent; i++) {
+			copies_sent = rcomm_cmd->copies_sent + rcomm_cmd->non_quorum_copies_sent;
+			for (i = 0; i < copies_sent; i++) {
 				if (rcomm_cmd->resp_list[i].status &
 				    (RECEIVED_OK|RECEIVED_ERR))
 					count++;
 			}
 
-			if (count == rcomm_cmd->copies_sent) {
+			if (count == copies_sent) {
```

Comment:
similar to this, in destroy_resp_list, memory need to be checked and freed for non_quorum_copies_sent indexes also

Reply:
even though data_ptr will be non-null for read IOs only and non-quorum replicas will not get read IOs, its better to check and free in destroy_resp_list